vhdbuilder/packer/install-dependencies.sh

```diff
@@ -404,16 +404,21 @@ K8S_VERSIONS="
 1.18.2.1
 "
 for PATCHED_KUBERNETES_VERSION in ${K8S_VERSIONS}; do
-  HYPERKUBE_URL="mcr.microsoft.com/oss/kubernetes/hyperkube:v${PATCHED_KUBERNETES_VERSION}"
-  # NOTE: the KUBERNETES_VERSION will be used to tag the extracted kubelet/kubectl in /usr/local/bin
-  # it should match the KUBERNETES_VERSION format(just version number, e.g. 1.15.7, no prefix v)
-  # in installKubeletAndKubectl() executed by cse, otherwise cse will need to download the kubelet/kubectl again
-  KUBERNETES_VERSION=$(echo ${PATCHED_KUBERNETES_VERSION} | cut -d"_" -f1 | cut -d"-" -f1 | cut -d"." -f1,2,3)
-  # extractHyperkube will extract the kubelet/kubectl binary from the image: ${HYPERKUBE_URL}
-  # and put them to /usr/local/bin/kubelet-${KUBERNETES_VERSION}
-  extractHyperkube "docker"
-  # remove hyperkube here as the one that we really need is pulled later
-  docker image rm $HYPERKUBE_URL
+  if (($(echo ${KUBERNETES_VERSION} | cut -d"." -f2) < 17)); then
+    HYPERKUBE_URL="mcr.microsoft.com/oss/kubernetes/hyperkube:v${PATCHED_KUBERNETES_VERSION}"
+    # NOTE: the KUBERNETES_VERSION will be used to tag the extracted kubelet/kubectl in /usr/local/bin
+    # it should match the KUBERNETES_VERSION format(just version number, e.g. 1.15.7, no prefix v)
+    # in installKubeletAndKubectl() executed by cse, otherwise cse will need to download the kubelet/kubectl again
+    KUBERNETES_VERSION=$(echo ${PATCHED_KUBERNETES_VERSION} | cut -d"_" -f1 | cut -d"-" -f1 | cut -d"." -f1,2,3)
```

Comment:
KUBERNETS_VERSION is also needed by extractKubeBinaries right? So it should be assigned before if-else

Reply:
extractKubeBinaries has it's own KUBERNETS_VERSION, but agree with your other comment that we should assign the variable by the caller, instead of in the extractKubeBinaries function